### PR TITLE
[MOSTLY MODULAR] Body Teleporter Medicell

### PR DESCRIPTION
**Pull request by softcerv**
## About The Pull Request

Adds in the Body Teleporter Medicell that shoots a projectile that teleports a dead body to the user with it if hit this only works on humans.

body_teleporter

Video of the cell in action

https://user-images.githubusercontent.com/68373373/142778981-5fab5ec2-9da0-42ce-8862-83104aa13d4c.mp4

Ingame Screenshot
image
## How This Contributes To The Skyrat Roleplay Experience

This gives paramedics an easier way to rescue bodies from crowded areas, especially if sec is being uncooperative. Having it require a bluespace slime core also means that medical will  have to interact with science more.
## Changelog

:cl:
expansion: Adds the Body Teleporter Medicell to the game.
/:cl:

### CODE REVIEW

**Comment by YakumoChen** [comment]
give these a check so you can't antag test fake death changelings with it please

**Comment by softcerv** [comment]
> give these a check so you can't antag test fake death changelings with it please

I'm not familiar with the code, how would you go about doing that?

**Comment by YakumoChen** [comment]
Check ling code if fake death grants a trait and add it to the argument

**Comment by softcerv** [comment]
> Check ling code if fake death grants a trait and add it to the argument

done

File: modular_skyrat/modules/modular_weapons/code/medguns.dm [hunk 1]
```diff
@@ -245,15 +245,27 @@
 	icon_state = "gown"
 	ammo_type = /obj/item/ammo_casing/energy/medical/utility/bed
 
+/obj/item/weaponcell/medical/utility/body_teleporter
+	name = "Body Transporter Medicell"
+	desc = "A medicell that allows the user to transport a dead body to themselves."
+	icon_state = "body" //IMPORT A CUSTOM TEXTURE LATER
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_weapons/code/medguns.dm]: Is this meant to still be there?

File: modular_skyrat/modules/modular_weapons/code/medguns.dm [hunk 2]
```diff
@@ -245,15 +245,27 @@
...
 	from_obj = /obj/item/food/grown/aloe
 	to_obj = /obj/item/weaponcell/medical/utility/salve
 
 /obj/item/device/custom_kit/empty_cell/Initialize()
 	. = ..()
 	AddElement(/datum/element/item_scaling, 0.5, 1)
+
+/obj/item/device/custom_kit/empty_cell/body_teleporter
+	name = "Empty Body Teleporter Medicell"
+	desc = "an unactivated Body Teleporter Medicell, use this on a bluespace slime extract to make this into a usable cell"
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_weapons/code/medguns.dm]: Item names are never capitalized unless it's a proper noun.
```suggestion
	name = "empty body teleporter medicell"
	desc = "An inactive body teleporter medicell, use this on a bluespace slime extract to make this into a usable cell."
```

File: modular_skyrat/modules/modular_weapons/code/medguns.dm [hunk 3]
```diff
@@ -245,15 +245,27 @@
...
 
+/obj/item/weaponcell/medical/utility/body_teleporter
+	name = "Body Transporter Medicell"
+	desc = "A medicell that allows the user to transport a dead body to themselves."
+	icon_state = "body" //IMPORT A CUSTOM TEXTURE LATER
+	ammo_type = /obj/item/ammo_casing/energy/medical/utility/body_teleporter
+
 //Empty Medicell//
 /obj/item/device/custom_kit/empty_cell //Having the empty cell as an upgrade kit sounds jank, but it should work well.
 	name = "Empty Salve Medicell"
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_weapons/code/medguns.dm]: Ditto.
```suggestion
	name = "empty salve medicell"
```

File: modular_skyrat/modules/modular_weapons/code/medguns.dm [hunk 4]
```diff
@@ -245,15 +245,27 @@
...
 	icon = 'modular_skyrat/modules/modular_weapons/icons/obj/upgrades.dmi'
 	icon_state = "empty"
-	desc = "A unactivated Salve Medicell, use this on an aloe leaf to make this into a usable cell"
+	desc = "an unactivated Salve Medicell, use this on an aloe leaf to make this into a usable cell"
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_weapons/code/medguns.dm]:
```suggestion
	desc = "An inactive salve medicell, use this on an aloe leaf to make this into a usable cell."
```

File: modular_skyrat/modules/modular_weapons/code/mediguns.dm [hunk 5]
```diff
@@ -141,6 +141,16 @@
 	category = list("Ammo")
 	departmental_flags = DEPARTMENTAL_FLAG_MEDICAL
 
+/datum/design/bodymedicell
+	name = "Empty Body Teleporter Medicell"
+	desc = "A Empty Medicell that can be upgraded by a bluespace slime extract into a usable Body Teleporter Medicell."
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_weapons/code/mediguns.dm]:
```suggestion
	desc = "An empty medicell that can be upgraded by a bluespace slime extract into an usable body teleporter medicell."
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
We're close.

**Review by GoldenAlpharex — APPROVED** [review]

**Pull request merged**